### PR TITLE
Prepare input bar for vision

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR refactors the input bar file upload to prepare for vision support. I implemented a custom hook to manage all file uploads, centralizing the logic in one place. Although the code already supports image uploads, it is currently unused as we do not yet accept images in the input.

image

### CODE REVIEW

File: front/hooks/useFileUploaderService.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            );
+          }
+
+          if (file.size > MAX_FILE_SIZE) {
+            return Promise.reject(
+              new FileBlobUploadError("file_too_large", file)
+            );
+          }
+
+          if (contentType.startsWith("image/")) {
```
Comment (JulesBelveze) [inline front/hooks/useFileUploaderService.ts:83]: Why not creating a `isSupportedImageContentFragmentType` for consistency with the above? Also I guess we're not planning on supporting all image type, right?

File: front/hooks/useFileUploaderService.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            const base64Text = await getPreview(file);
+
+            return createFileBlob(file, base64Text, contentType, base64Text);
+          }
+
+          if (contentType === "application/pdf") {
```
Comment (JulesBelveze) [inline front/hooks/useFileUploaderService.ts:89]:
```suggestion
          else if (contentType === "application/pdf") {
```

File: front/hooks/useFileUploaderService.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
+import type { SupportedContentFragmentType } from "@dust-tt/types";
+import { isSupportedTextContentFragmentType } from "@dust-tt/types";
+import { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { extractTextFromPDF } from "@app/lib/client/handle_file_upload";
+import { getMimeTypeFromFile } from "@app/lib/file";
+
+interface FileContent {
```
Comment (JulesBelveze) [inline front/hooks/useFileUploaderService.ts:9]: **Question:** 
Kindda feels like we are starting to have a lot of different types to handle Files/ContentFragment makes it hard to understand. WDYT? Dunno if we could have `UploadedContentFragment` or `FileContent`?

File: front/hooks/useFileUploaderService.ts (hunk 2 above)
Comment (JulesBelveze) [reply front/hooks/useFileUploaderService.ts:89]: `nit` but I feel else if makes it easier to understand

File: front/hooks/useFileUploaderService.ts (hunk 1 above)
Comment (flvndvd) [reply front/hooks/useFileUploaderService.ts:83]: It's coming in the follow up PR as I need it widely in the BE.

File: front/hooks/useFileUploaderService.ts (hunk 2 above)
Comment (flvndvd) [reply front/hooks/useFileUploaderService.ts:89]: Yep 💯 forgot after refactoring.

File: front/hooks/useFileUploaderService.ts (hunk 3 above)
Comment (flvndvd) [reply front/hooks/useFileUploaderService.ts:9]: The goal of this initiative is too split file upload from content fragment.

**Review by JulesBelveze — APPROVED** [review]

**Pull request merged**